### ISSUE TITLE
Freeze

### ISSUE THREAD

**Issue by adeftu** [KronicDeth/intellij-elixir#2788]
# System

Plugin Version: 12.2.1
Application: IntelliJ IDEA Ultimate Edition (2021.3.3)
Operating System: Mac OS X (12.5)

[threadDumps-freeze-20220810-135216-IU-213.7172.25.zip](https://github.com/KronicDeth/intellij-elixir/files/9302835/threadDumps-freeze-20220810-135216-IU-213.7172.25.zip)
[threadDumps-freeze-20220810-134630-IU-213.7172.25.zip](https://github.com/KronicDeth/intellij-elixir/files/9302837/threadDumps-freeze-20220810-134630-IU-213.7172.25.zip)

**Comment by sh41** [comment]
Duplicate of #2127. Fixed in v23.2.0 (2026-05-15).

Both attached `threadDumps-freeze-*` zips diagnose it: 36 samples each over 180 seconds, and every one of the 72 shows two `Action Updater (Common)` threads RUNNABLE in `UnaliasedName.up`, reached from `UsageTargetProvider` -> `PsiElement2UsageTargetAdapter` -> `ItemPresentation.getPresentableText`. In 12.2.1 that function's `QualifiedMultipleAliases` branch passed the same element back to itself, so resolving the qualifier in `alias Prefix.{Suffix}` looped forever. It is `tailrec`, so it spins rather than overflowing -- a hang, not a crash. The spinning threads hold read locks, which is why the write actions you were waiting on never started: opening a file from the Project tree in one dump, an autosave in the other. That is why it presented as a bare freeze with no obvious trigger.

Fixed in a056a8b666f (#3818) and released in v23.2.0, with regression tests in `ItemPresentationTest` and `UnaliasedNameTest` asserting termination on this shape.

Closed as part of an AI-assisted review of the backlog. A human checked this one before closing it, but it might still be a mistake. If you think the reason above is wrong, please reopen the issue and tell us why.
